### PR TITLE
Sort comments by content

### LINKED ISSUE #1401: Improve comment sorting
Hi guys,

the sort button in the comments panel currently only sorts comments within the respective treeIDs, but does not affect the ordering of the actual trees. 

For nmls containing multiple trees with meaningful IDs it would be handy to have the trees alphabetically sorted as well. 

Here is an example file for that:

https://webknossos.brain.mpg.de/annotations/Explorational/579f92075d00003342a4a202#20912,27327,513,0,0.16,19182

I'd like to have the sort button arrange the trees like that:

> brain em
> brain lm
> donald em
> donald em
> ....

Thanks for solving this!

Florian

florian.drawitsch@brain.mpg.de

 Reported by: Florian Drawitsch (florian@mhlab.net) on master

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/1401/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by hotzenklotz**
### Mailable description of changes:
- Changed comment sorting to sort based on content not id using natural sort order

### Steps to test:
- abc

### Issues:
- fixes #1401 

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/libs/utils.js [hunk 1]
```diff
@@ -94,6 +102,16 @@ const Utils = {
     };
   },
 
+  localeCompareBy<T: Object>(key: string, isSortedAscending: boolean = true) {
+    const sortingOrder = isSortedAscending ? 1 : -1;
+
+    return (a: T, b: T) =>
+      a[key].localeCompare(b[key], "en", {
+        numeric: true,
```
Comment (daniel-wer) [inline app/assets/javascripts/libs/utils.js]: :heart:

**Comment by daniel-wer** [comment]
Also please commit your `yarn.lock` file :)

**Comment by daniel-wer** [comment]
Works like a charm. By the way why did you base this PR on the `user-list-react` branch?
Can be merged once the e2e tests are dealt with.

**Comment by hotzenklotz** [comment]
@daniel-wer Because it uses the same sorting as the user list.

**Pull request merged**